### PR TITLE
twentyFirst, signUp activity & other helper class added

### PR DESCRIPTION
**Pull request by csabdulahad**
First of all, i moved the welcome activity code to the main activity. Because we need to do a lot of things in the welcome activity. Such as showing brand logo on app start, showing special images and info on special days(21 Feb, 26 Mar, 16 Dec), showing signup options and so on.

Few helper classes have been added too. Just go through them you will understand their motives and capabilities. For **TwentyFirst**(showing images & info on special days of BD) project, i have added three images as you will see. Layout files now can use ConstraintLayout, as i made a change in the app level gradle file.

### CODE REVIEW

File: app/src/main/java/tk/dehoghori/dehoghori/helper/InputChecker.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        return true;
+    }
+
+    /**
+     * It checks whether a radio group is selected or not
+     *
+     * @param radioGroup which you want to check for selection
+     * @return false if no radio button is selected, true if one radio button is selected
+     */
+    public static boolean passSelectionCheck(RadioGroup radioGroup) {
```
Comment (Zia005) [inline app/src/main/java/tk/dehoghori/dehoghori/helper/InputChecker.java]: Why should we use it

File: app/src/main/java/tk/dehoghori/dehoghori/helper/InputChecker.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+     */
+    public static boolean passTextSpaceCheck(EditText editText) {
+        String data = editText.getText().toString();
+        if (data.isEmpty()) {
+            editText.setError("Input is empty");
+            editText.requestFocus();
+            return false;
+        }
+
+        if (!Pattern.matches("[a-zA-Z\\s]+", data)) {
```
[the remark this answers is not in the document]
Comment (csabdulahad) [reply app/src/main/java/tk/dehoghori/dehoghori/helper/InputChecker.java]: this is from regEx(Regular Expression). **\s** means space. with that regEd i meant to say, that accept characters from a-z, A-z and spaces. hope you get it.

File: app/src/main/java/tk/dehoghori/dehoghori/helper/InputChecker.java (hunk 1 above)
Comment (csabdulahad) [inline app/src/main/java/tk/dehoghori/dehoghori/helper/InputChecker.java]: @Zia005 

**wait! did you checkout this repository as an android project in android studio and run the app in an emulator ? answer this first. Please. If you don't then you should do it by now. don't go ahead and try to learn code this way. you should first run the app, then you read the doc.**

a radio group consists of two/more radio buttons right ? we use the method to see whether the user has at least chosen one option/radioButton from that radio group. you will get to see how this function reacts, when you are in the sign up activity, and you don't choose your gender, then you try to say get started. you would see that it immediately catches your attention and animate the radio group to hint you that you missed this input.

Comment (csabdulahad) [reply app/src/main/java/tk/dehoghori/dehoghori/helper/InputChecker.java]: i just mistakenly deleted your review here!

**Pull request merged**